### PR TITLE
Updating data for a push

### PR DESCRIPTION
**Pull request by AlasdairSwan**
@rlucioni @jimabramson here is the update to make it so that PATCH only sends the changed data. Thus involved less code than I first thought.

### CODE REVIEW

**Comment by jimabramson** [comment]
nice :+1:

File: programs/static/js/models/program_model.js [hunk 1]
```diff
@@ -70,6 +70,10 @@ define([
                  */
                 method = this.isNew() ? 'create' : (patch ? 'patch' : 'update');
 
+                if ( patch ) {
+                    config.data = JSON.stringify( options.update ) || this.attributes;
```
Comment (rlucioni) [inline programs/static/js/models/program_model.js]: Looks like this is the line causing coverage to fail.

Comment (AlasdairSwan) [inline programs/static/js/models/program_model.js]: Yeah, I am just pulling out the config into a getConfig function that I can write tests for

Comment (AlasdairSwan) [inline programs/static/js/models/program_model.js]: And makes the code neater as an added bonus.

File: programs/static/js/test/specs/program_details_spec.js [hunk 2]
```diff
@@ -241,6 +241,27 @@ define([
...
+
+                it( 'should create a POST config object by default', function() {
+                    var config = view.model.getConfig();
+
+                    expect( config.type ).toEqual( 'POST' );
+                    expect( config.contentType ).toEqual( 'application/json' );
+                    expect( config.data ).not.toBeDefined();
+                });
+
+                it( 'should create a PATCH config object by default', function() {
```
Comment (rlucioni) [inline programs/static/js/test/specs/program_details_spec.js]: I'd strike the "by default" part. It should create a PATCH config object when patching is explicitly requested.

**Comment by rlucioni** [comment]
Just a minor comment about tests, otherwise :+1: .

**Pull request merged**